R/responsibilities.R

```diff
@@ -0,0 +1,318 @@
+                                        touchstone_name, location = NULL) {
+  assert_scalar_character(touchstone_name)
+  res <- helper_touchstones(modelling_group_id, location)
+
+  if (length(res)==0) {
+    stop(sprintf("Unknown touchstone with id '%s'", touchstone_name))
+
+  } else if (sum(which(vcapply(res, "[[", "id") == touchstone_name))>0) {
+    versions <- res[[which(vcapply(res, "[[", "id") == touchstone_name)]]$versions
+    data_frame(
+      id = vcapply(versions, "[[", "id"),
+      name = vcapply(versions, "[[", "name"),
+      version = viapply(versions, "[[", "version"),
+      description = vcapply(versions, "[[", "description"),
+      status = vcapply(versions, "[[", "status"))
+
+  } else {
+    stop(sprintf("Unknown touchstone with id '%s'", touchstone_name))
+  }
+}
+
+helper_get_touchstone <- function(modelling_group_id, touchstone_id,
+                                  location = NULL) {
+  assert_scalar_character(modelling_group_id)
+  assert_scalar_character(touchstone_id)
+  path <- sprintf("/modelling-groups/%s/responsibilities/%s/",
+                  modelling_group_id, touchstone_id)
+  montagu_api_GET(location, path)
+}
+
+helper_get_responsibility <- function(modelling_group_id, touchstone_id,
+                                      scenario_id, location = NULL) {
+
+  resps <- helper_get_touchstone(modelling_group_id, touchstone_id,
+                                 location)$responsibilities
+
+  select <- which(vcapply(resps, function(x) x$scenario$id) == scenario_id)
+  if (length(select)==0) {
+    stop(sprintf("Unknown scenario with id '%s'", scenario_id))
+  }
+  resps[[select]]
+}
+
+##' This may get a name change pending splitting apart of various
+##' montagu components.
+##' @title Retrieve information about a scenario
+##' @param touchstone_id Id of the touchstone (includes version suffix)
+##' @inheritParams montagu_touchstone_versions
+##' @return Data frame of scenario_id, description and disease.
+##' @export
+montagu_scenarios <- function(modelling_group_id,
+                              touchstone_id, location = NULL) {
+  resps <- helper_get_touchstone(modelling_group_id, touchstone_id,
+                                 location)$responsibilities
+  data_frame(
+    scenario_id = vcapply(resps, function(x) x$scenario$id),
+    description = vcapply(resps, function(x) x$scenario$description),
+    disease = vcapply(resps, function(x) x$scenario$disease))
+}
+
+##' This may get a name change pending splitting apart of various
+##' montagu components.
+##' @title Retrieve current status of a groups' scenario.
+##' @param scenario_id Id of the scenario
+##' @inheritParams montagu_scenarios
+##' @return "invalid", "complete", "valid", or "empty"
+##' @export
+montagu_scenario_status <- function(
+  modelling_group_id, touchstone_id, scenario_id, location = NULL) {
+
+  helper_get_responsibility(modelling_group_id, touchstone_id, scenario_id,
+                      location)$status
```

Comment:
indentation - looks like the whole file could benefit from a re-indent

Reply:
Try now - I've restyled a bit.

Reply:
The long param and function names sometimes make it a bit messy.